### ISSUE TITLE
List index out of range Error when trying to generate a model properties

### ISSUE THREAD

**Issue by BlueSeaDragon** [bigmoostache/geneticNN#1]
The exact error is:
 
```
Traceback (most recent call last):
File ~/DataspellProjects/ML-Projects/geneticNN/src/geneticalgorithm.py:165, in GeneticAlgorithm.generate_candidates(self, generation, number_of_candidates)
    159     new_model_skeleton = structure_modifiers[starting_id].remove_run(model=[starting_model.model_skeleton,
    160                                                                             starting_model.model_properties],
    161                                                                      run_to_remove=proposed_run,
    162                                                                      replacement_pairings=proposed_pairings,
    163                                                                      inplace=False)
    164 new_model = Model(model_name=f"model_{generation}_{k}", model_skeleton=new_model_skeleton)
--> 165 new_model.model_properties = ModelPropertiesMapper(new_model_skeleton)
    166 new_model.model_parameters = self.resolve_parameters(starting_model, new_model)
    167 new_model.model_parameters = self.random_update_parameters(new_model)

File ~/DataspellProjects/ML-Projects/geneticNN/src/modelproperties.py:135, in ModelPropertiesMapper.__init__(self, model_skeleton, detailed_props)
    132 self.sub_props: dict = None
    133 self.sympy_data: dict = {}
--> 135 self.generate_properties()
    136 self.generate_constraints()
    137 self.generate_global_parameters()

File ~/DataspellProjects/ML-Projects/geneticNN/src/modelproperties.py:165, in ModelPropertiesMapper.generate_properties(self)
    163 outputs = self.model_skeleton.outputs
    164 for output_var, output_info in outputs.items():
--> 165     output_model = self.model_skeleton.runs[output_info[0]]["id"]
    166     output_var_name = output_info[1]
    167     variables[output_var] = self.sub_props[output_model]["variables"][output_var_name]

IndexError: list index out of range
```

It comes from a variation of 
```JSON
{
    "inputs": {"X": {"dim": 1}},
    "outputs": {"Y": [0, "Y"]},
    "runs": [{"id": "Linear_4", "inputs": {"X": [-1, "X"]}}],
    "submodels": {
        "Linear_2": {"source": "basic_templates", "type": "Linear"},
        "Linear_3": {"source": "basic_templates", "type": "Linear"},
        "Linear_4": {"source": "basic_templates", "type": "Linear"}
    }
}
```
or
```JSON
{
    "inputs": {"X": {"dim": 1}},
    "outputs": {"Y": [2, "Y"]},
    "runs": [
        {"id": "Linear_2", "inputs": {"X": [-1, "X"]}},
        {"id": "Linear_3", "inputs": {"X": [0, "Y"]}},
        {"id": "ReLU_0", "inputs": {"X": [1, "Y"]}}
    ],
    "submodels": {
        "Linear_2": {"source": "basic_templates", "type": "Linear"},
        "Linear_3": {"source": "basic_templates", "type": "Linear"},
        "ReLU_0": {"source": "basic_templates", "type": "ReLU"}
    }
}
```

I suspect a remove from the first one -> runs gets empty which leads to errors

**Comment by BlueSeaDragon** [comment]
confirmed this is the error, ModelPropertiesMapper cannot handle a model_skeleton with no runs and output connected to input

**Comment by BlueSeaDragon** [comment]
Solved with the  commit 1b729d1
